app/utils/get_emails.py

```diff
@@ -0,0 +1,177 @@
+    """
+    Function to download data
+    :param token: slack api token or jwt
+    :param query_params: dict of query strings for the url
+    :param url: url
+    :return: data - json data gotten
+    """
+    logging.info('Requesting for the data...')
+    if url:
+        params = append_params({'token': token}, query_params)
+        res = requests.get(url, params=params)
+        if res.status_code == 200:
+            data = json.loads(res.content)
+            if data.get('ok'):
+                logging.info('Finished downloading data')
+                return data
+            else:
+                msg = data.get('error')
+                raise Exception(msg)
+        else:
+            msg = 'Site down or invalid URL'
+            raise Exception(msg)
+    else:
+        msg = 'URL cannot be None'
+        raise requests.exceptions.MissingSchema(msg)
+
+
+def get_people_data(person_ids, all_persons_data):
+    """
+    Function to get selected info from each person object
+
+    :param person_ids: List of ids of persons whose data we need
+    :param all_persons_data: list of person data objects
+    :return: persons_data - list of selected person data objects
+    """
+    logging.info('Selecting Names and email only from the downloaded data')
+    persons_data = []
+    for person_obj in all_persons_data:
+        if person_obj.get('id') in person_ids:
+            person_details = person_obj.get("profile")
+            # catch bots since they got no email
+            email = person_details.get('email')
+            if email:
+                selected_data = {
+                    'first_name': person_details.get('first_name'),
+                    'last_name': person_details.get('last_name'),
+                    'real_name': person_details.get('real_name'),
+                    'email': email
+                }
+                persons_data.append(selected_data)
+    logging.info('Selected data for {} People'.format(len(persons_data)))
+    return persons_data
+
+
+def get_peoples_id(group_data):
+    """
+    Function to return all persons whose location is similar
+    to the given location
+
+    :param group_data: Data of all persons
+    :return: persons_id - list of all group members
+    """
+    logging.info('Getting persons IDs')
+    persons_id = group_data.get('members')
+    logging.info('{} IDs found'.format(len(persons_id)))
+    return persons_id
+
+
+def write_to_file(persons_data, filename):
+    """
+    Function to write json data to file
+    :param persons_data: json data to be writen to file
+    :param filename: Name of the file the data is to be written to
+    """
+    logging.info('Starting to write the selected data to emails.json')
+    with open(filename, 'w') as f:
+        # ensure_ascii makes sure that all ascii characters are left untouched
+        json.dump(persons_data, f, ensure_ascii=False, indent=4)
+    logging.info('File writen --> Done')
+
+
+def request_people_data():
```

Comment:
I'd recommend you use the slack_client interface to make calls to the api that's been initialized in `app/__init__` over building the requests yourself.

Reply:
Fixed. I am now using the slack_client interface